### PR TITLE
allow to configure tracker via static methods + do not track on background

### PR DESCRIPTION
**Pull request by marcmrf**

### CODE REVIEW

File: CompassSDK/Tick/SendTik.swift [hunk 1]
```diff
@@ -18,7 +18,7 @@ struct TikApiCall: ApiCall {
     let baseUrl: URL?
     let type: ContentType
     
-    init(baseUrl: URL? = Bundle.main.compassEndpoint, params: [String : Any], path: String, type: ContentType) {
+    init(baseUrl: URL? = TrackingConfig.shared.endpoint, params: [String : Any], path: String, type: ContentType) {
```
Comment (marcmrf) [inline CompassSDK/Tick/SendTik.swift:21]: use Global config new class in order to be able to configure tracker via info.plist or static method

File: CompassSDK/Tracker/Core/CompassTracker.swift [hunk 2]
```diff
@@ -341,3 +358,27 @@ private extension CompassTracker {
...
+        stopObserving()
+    }
+    
+    func onAppActive(){
+        doTik()
+    }
+    
+    func configureAppLifecycleListeners() {
+        NotificationCenter.default.addObserver(forName: UIApplication.didBecomeActiveNotification, object: nil, queue: .main) { _ in
+            self.onAppActive()
```
Comment (marcmrf) [inline CompassSDK/Tracker/Core/CompassTracker.swift:373]: recover tracking

File: CompassSDK/Tracker/Core/CompassTracker.swift [hunk 3]
```diff
@@ -341,3 +358,27 @@ private extension CompassTracker {
...
+        }
+
+        NotificationCenter.default.addObserver(forName: UIApplication.willResignActiveNotification, object: nil, queue: .main) { _ in
+            self.onAppInactive()
```
Comment (marcmrf) [inline CompassSDK/Tracker/Core/CompassTracker.swift:377]: on app change app preview, aka carousel for selecting other open app

File: CompassSDK/Tracker/Core/CompassTracker.swift [hunk 4]
```diff
@@ -341,3 +358,27 @@ private extension CompassTracker {
...
+        }
+
+        NotificationCenter.default.addObserver(forName: UIApplication.didEnterBackgroundNotification, object: nil, queue: .main) { _ in
+            self.onAppInactive()
```
Comment (marcmrf) [inline CompassSDK/Tracker/Core/CompassTracker.swift:381]: on pureley background state

File: CompassSDK/Tracker/Core/CompassTracker.swift [hunk 5]
```diff
@@ -329,6 +345,7 @@ private extension CompassTracker {
         stopObserving()
         operationQueue.cancelAllOperations()
         trackInfo.pageUrl = pageName
+        trackInfo.tik = 0
```
Comment (marcmrf) [inline CompassSDK/Tracker/Core/CompassTracker.swift:348]: tick was not reset between pageViews

**Review by miquelmasrieramrf — APPROVED** [review]

**Pull request merged**